### PR TITLE
Adding inline filter support

### PR DESCRIPTION
**Pull request by renebrandel**
#### Description of changes:

Adds support to do inline content filters.

### Instructions

See pages `src/pages/lib/q/platform/[platform].mdx` for instructions to play with inline filters

**If this PR should not be merged upon approval for any reason, please submit as a DRAFT**

Which product(s) are affected by this PR (if applicable)?
- [ ] amplify-cli
- [ ] amplify-ui
- [ ] amplify-studio
- [ ] amplify-hosting
- [ ] amplify-libraries

Which platform(s) are affected by this PR (if applicable)?
- [ ] JS
- [ ] iOS
- [ ] Android
- [ ] Flutter
- [ ] React Native

**Please add the product(s)/platform(s) affected to the PR title**

#### Checks

- [ ] Does this PR conform to [the styleguide](https://github.com/aws-amplify/docs/blob/main/STYLEGUIDE.md)?

- [ ] Does this PR include filetypes other than markdown or images? Please add or update unit tests accordingly.

- [ ] Are any files being deleted with this PR? If so, have the needed redirects been created?

- [ ] Are all links in MDX files using the MDX link syntax rather than HTML link syntax? <br /> 
      _ref: MDX: `[link](https://link.com)` 
            HTML: `<a href="https://link.com">link</a>`_

### When this PR is ready to merge, please check the box below
- [ ] Ready to merge

_By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license._

### CODE REVIEW

File: src/components/InlineFilter/index.tsx [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+type InlineFilterProps = {
+  children: JSX.Element
+  filters?: string[]
+  filter?: string
+}
+
+export default function InlineFilter({ filter, filters, children }: InlineFilterProps) {
+  if (filter && filters) {
+    throw new Error("Can't set both `filter` and `filters` props.")
```
Comment (reesscot) [inline src/components/InlineFilter/index.tsx:12]: Do we want to throw here? If someone does this by mistake and we don't catch it in PR, won't this whitescreen the page? cc @katiegoines

File: src/components/InlineFilter/__tests__/InlineFilter.test.tsx [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+      };
+    };
+
+    render(<InlineFilter filters={["js", "react-native"]}><span>{inlineContent}</span></InlineFilter>)
+
+    const inlineFilterNode = await screen.findByText(inlineContent);
+    expect(inlineFilterNode).toBeInTheDocument();
+  });
+
+  it(`shouldn't render the inilne content when using an invalid value with multiple filters`, async () => {
```
Comment (reesscot) [inline src/components/InlineFilter/__tests__/InlineFilter.test.tsx:41]: nit
```suggestion
  it(`shouldn't render the inline content when using an invalid value with multiple filters`, async () => {
```

File: src/components/InlineFilter/__tests__/InlineFilter.test.tsx [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+  it(`shouldn't render when only passing an empty array to filters`, async () => {
+    routerMock.useRouter = () => {
+      return {
+        query: {
+          platform: "js"
+        }
+      };
+    };
+
+    expect(() => render(<InlineFilter filters={[]}><span>{inlineContent}</span></InlineFilter>)).toThrow()
```
Comment (reesscot) [inline src/components/InlineFilter/__tests__/InlineFilter.test.tsx:64]: this needs to be updated to check for empty div?

File: src/components/InlineFilter/index.tsx [hunk 4]
```diff
@@ -0,0 +1,21 @@
+import type React from 'react';
+import FilterChildren from '../FilterChildren';
+
+type InlineFilterProps = {
+  children: JSX.Element
+  filters: string[]
+}
+
+export default function InlineFilter({ filters, children }: InlineFilterProps) {
+  if (!filters || filters.length < 1) {
```
Comment (reesscot) [inline src/components/InlineFilter/index.tsx:10]:
```suggestion
  if (!filters || !Array.isArray(filters) || filters.length < 1) {
```

File: src/components/InlineFilter/__tests__/InlineFilter.test.tsx (hunk 3 above)
Comment (renebrandel) [reply src/components/InlineFilter/__tests__/InlineFilter.test.tsx]: Addressed.

File: src/components/InlineFilter/index.tsx [hunk 5]
```diff
@@ -0,0 +1,21 @@
+import type React from 'react';
+import FilterChildren from '../FilterChildren';
+
+type InlineFilterProps = {
+  children: JSX.Element
```
Comment (reesscot) [inline src/components/InlineFilter/index.tsx:5]:
```suggestion
  children: React.ReactNode
```

https://github.com/DefinitelyTyped/DefinitelyTyped/blob/d076add9f29db350a19bd94c37b197729cc02f87/types/react/index.d.ts#L822

**Review by katiegoines — APPROVED** [review]

**Review by reesscot — APPROVED** [review]
LGTM!

**Pull request merged**